onmt/models/model.py

```diff
@@ -75,3 +93,70 @@ def count_parameters(self, log=print):
             log('decoder: {}'.format(dec))
             log('* number of parameters: {}'.format(enc + dec))
         return enc, dec
+
+
+class LanguageModel(BaseModel):
+    """
+    Core trainable object in OpenNMT. Implements a trainable interface
+    for a simple, generic  decoder model.
+    Args:
+      decoder (onmt.decoders.DecoderBase): a decoder object
+    """
+
+    def __init__(self, encoder=None, decoder=None):
+        super(LanguageModel, self).__init__(encoder, decoder)
+        if encoder is not None:
+            raise ValueError("LanguageModel should not be used"
+                             "with an encoder")
+        self.decoder = decoder
+
+    def forward(self, src, tgt, lengths, bptt=False, with_align=False):
+        """Forward propagate a `src` and `tgt` pair for training.
+        Possible initialized with a beginning decoder state.
+        Args:
+            src (Tensor): A source sequence passed to decoder.
+                typically for inputs this will be a padded `LongTensor`
+                of size ``(len, batch, features)``. However, may be an
+                image or other generic input depending on decoder.
+            tgt (LongTensor): A target sequence passed to decoder.
+                Size ``(tgt_len, batch, features)``.
+            lengths(LongTensor): The src lengths, pre-padding ``(batch,)``.
+            bptt (Boolean): A flag indicating if truncated bptt is set.
+                If reset then init_state
+            with_align (Boolean): A flag indicating whether output alignment,
+                Only valid for transformer decoder.
+        Returns:
+            (FloatTensor, dict[str, FloatTensor]):
+            * decoder output ``(tgt_len, batch, hidden)``
+            * dictionary attention dists of ``(tgt_len, batch, src_len)``
+        """
+
+        if not bptt:
+            self.decoder.init_state()
+        dec_out, attns = self.decoder(
+            src, memory_bank=None, memory_lengths=lengths,
+            with_align=with_align
+        )
+        return dec_out, attns
+
+    def update_dropout(self, dropout):
+        self.decoder.update_dropout(dropout)
+
+    def count_parameters(self, log=print):
+        """Count number of parameters in model (& print with `log` callback).
+        Returns:
+            (int, int):
+            * encoder side parameter count
+            * decoder side parameter count
+        """
+
+        enc, dec = 0, 0
+        for name, param in self.named_parameters():
+            if "decoder" in name:
+                dec += param.nelement()
+
+        if callable(log):
+            log("encoder: {}".format(enc))
```

Comment:
?

Reply:
No encoder in LM but I chose to keep it in the count to make it clear. Do you want it removed ?

Reply:
Maybe just add a comment to underline the fact it's intentional.